sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1004,6 +1003,15 @@ DynRTDeviceBinaryImage::DynRTDeviceBinaryImage(
   Bin->EntriesBegin = nullptr;
   Bin->EntriesEnd = nullptr;
   Bin->Format = pi::getBinaryImageFormat(Bin->BinaryStart, DataSize);
+
+  switch (Bin->Format) {
+  case PI_DEVICE_BINARY_TYPE_SPIRV:
+    Bin->DeviceTargetSpec = PI_DEVICE_BINARY_TARGET_SPIRV64;
+    break;
+  default:
+    Bin->DeviceTargetSpec = PI_DEVICE_BINARY_TARGET_UNKNOWN;
+  }
```

Comment:
Does this work with the CUDA backend when there are multiple binaries?

Reply:
i think you can only load one `DynRTDeviceBinaryImage` at a time using the `SYCL_USE_KERNEL_SPV` env-variable. this just sets the devicetarget to spir64 in case it is a spir-file so the opencl backend can recognize it.